avocado/core/suite.py

```diff
@@ -89,21 +91,23 @@ def _from_config_with_resolver(cls, config, name=None):
                    config=config,
                    tests=tasks)
 
-    def _parse_variant(self):
-        # Varianter not yet parsed, apply configs
-        if not self.variant.is_parsed():
-            try:
-                self.variant.parse(self.config)
-            except (IOError, ValueError) as details:
-                raise OptionValidationError("Unable to parse "
-                                            "variant: %s" % details)
-
     @property
     def references(self):
         if self._references is None:
             self._references = self.config.get('run.references')
         return self._references
 
+    @property
+    def runner(self):
+        if self._runner is None:
+            runner_name = self.config.get('run.test_runner') or 'runner'
+            try:
+                runner_extension = RunnerDispatcher()[runner_name]
+                self._runner = runner_extension.obj
+            except KeyError:
```

Comment:
This goes against the description of this commit being a refactor, because it now won't return silently when the selected runner doesn't exist (which I think is a positive change).

Also, this commit has some noise in the commit message (seems like left over comment and duplicated signed off).

Reply:
@clebergnu, nIce catch, thanks. Just force-pushed here, since it was just a small change on the commit log message.